diff --git a/metadata.c b/metadata.c
index 4781db7..b29e8e1 100644
--- a/metadata.c
+++ b/metadata.c
@@ -1544,9 +1544,14 @@ video_no_dlna:
 
 	if( !m.date )
 	{
-		m.date = malloc(20);
+		m.date = malloc(21);
 		modtime = localtime(&file.st_mtime);
-		strftime(m.date, 20, "%FT%T", modtime);
+		/*
+		 * This is NOT a correct ISO-8610 format because modtime is a localtime.
+		 * However, at least, Kodi and Regza ZG2 work fine with this format.
+		 * Probably, these devices ignore the timezone in dc:date.
+		 */
+		strftime(m.date, 21, "%FT%TZ", modtime);
 	}
 
 	if( !m.title )
